src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -0,0 +1,880 @@
+  RAY_CHECK_OK(worker2.Objects().Wait(all_ids, 2, -1, &wait_results));
+  ASSERT_EQ(wait_results.size(), 3);
+  ASSERT_EQ(wait_results, std::vector<bool>({true, true, false}));
+
+  RAY_CHECK_OK(worker2.Objects().Wait(all_ids, 3, 100, &wait_results));
+  ASSERT_EQ(wait_results.size(), 3);
+  ASSERT_EQ(wait_results, std::vector<bool>({true, true, false}));
+
+  // Test Delete() from all machines.
+  // clear the reference held by PlasmaBuffer.
+  results.clear();
+  RAY_CHECK_OK(worker2.Objects().Delete(ids, false, false));
+
+  // Note that Delete() calls RayletClient::FreeObjects and would not
+  // wait for objects being deleted, so wait a while for plasma store
+  // to process the command.
+  usleep(1000 * 1000);
+  // Verify objects are deleted from both machines.
+  RAY_CHECK_OK(worker2.Objects().Get(ids, 0, &results));
+  ASSERT_EQ(results.size(), 2);
+  ASSERT_TRUE(!results[0]);
+  ASSERT_TRUE(!results[1]);
+
+  RAY_CHECK_OK(worker1.Objects().Get(ids, 0, &results));
+  ASSERT_EQ(results.size(), 2);
+  ASSERT_TRUE(!results[0]);
+  ASSERT_TRUE(!results[1]);
+}
+
+TEST_F(SingleNodeTest, TestNormalTaskLocal) {
+  std::unordered_map<std::string, double> resources;
+  TestNormalTask(resources);
+}
+
+TEST_F(TwoNodeTest, TestNormalTaskCrossNodes) {
+  std::unordered_map<std::string, double> resources;
+  resources.emplace("resource1", 1);
+  TestNormalTask(resources);
+}
+
+TEST_F(SingleNodeTest, TestActorTaskLocal) {
+  std::unordered_map<std::string, double> resources;
+  TestActorTask(resources, false);
+}
+
+TEST_F(TwoNodeTest, TestActorTaskCrossNodes) {
+  std::unordered_map<std::string, double> resources;
+  resources.emplace("resource1", 1);
+  TestActorTask(resources, false);
+}
+
+TEST_F(SingleNodeTest, TestDirectActorTaskLocal) {
+  std::unordered_map<std::string, double> resources;
+  TestActorTask(resources, true);
+}
+
+TEST_F(TwoNodeTest, TestDirectActorTaskCrossNodes) {
+  std::unordered_map<std::string, double> resources;
+  resources.emplace("resource1", 1);
+  TestActorTask(resources, true);
+}
+
+TEST_F(SingleNodeTest, TestDirectActorTaskLocalFO) {
+  std::unordered_map<std::string, double> resources;
+  TestActorFO(resources, true);
+}
+
+TEST_F(TwoNodeTest, TestDirectActorTaskCrossNodesFO) {
+  std::unordered_map<std::string, double> resources;
+  resources.emplace("resource1", 1);
+  TestActorFO(resources, true);
+}
+
+TEST_F(SingleNodeTest, TestDirectActorTaskLocalFailure) {
+  std::unordered_map<std::string, double> resources;
+  TestActorFailure(resources, true);
+}
+
+TEST_F(TwoNodeTest, TestDirectActorTaskCrossNodesFailure) {
+  std::unordered_map<std::string, double> resources;
+  resources.emplace("resource1", 1);
+  TestActorFailure(resources, true);
+}
+
+TEST_F(SingleNodeTest, TestCoreWorkerConstructorFailure) {
+  try {
+    CoreWorker core_worker(WorkerType::DRIVER, Language::PYTHON, "",
+                           raylet_socket_names_[0], JobID::FromInt(1), gcs_options_,
+                           nullptr);
```

Comment:
we should add a `ASSERT_TRUE(false);` here to make sure the exception is really thrown.